api4/team.go

```diff
@@ -1485,6 +1485,29 @@ func inviteGuestsToChannels(c *Context, w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	// restrict guest invites depending on the subscription type (see MM-47228)
+	isCloud := c.App.Srv().License() != nil && *c.App.Srv().License().Features.Cloud
```

Comment:
I think we should instead check for the feature, e.g. `if c.App.Channels().License() != nil && *c.App.Channels().License().Features.GuestAccounts` . That way, both cloud and self hosted are covered. I think it was just a miss that we never added that check. Then I think we can omit all the subscription and product checking.